nuxeo-core/nuxeo-core-storage-mongodb/src/main/java/org/nuxeo/ecm/core/storage/mongodb/MongoDBRepository.java

```diff
@@ -345,7 +345,7 @@ public void createStates(List<State> states) {
             // Avoid hiding any others bulk errors
             if (duplicates.size() == mbwe.getWriteErrors().size()) {
                 log.trace("MongoDB:    -> DUPLICATE KEY: {}", duplicates);
-                ConcurrentUpdateException concurrentUpdateException = new ConcurrentUpdateException();
+                ConcurrentUpdateException concurrentUpdateException = new ConcurrentUpdateException("Concurrent update");
```

Comment:
is the message useful, or could it be richer?

Reply:
The lines below this one call `addInfo` with the details of the duplicates. I'm adding `"Concurrent update"` here to be in sync in with what's done in VCS and have unit tests that work for both DBS and VCS.